src/hotspot/os/aix/os_aix.cpp

```diff
@@ -3033,29 +3049,134 @@ void os::jfr_report_memory_info() {}
 
 // Simulate the library search algorithm of dlopen() (in os::dll_load)
 int os::Aix::stat64x_via_LIBPATH(const char* path, struct stat64x* stat) {
-  if (path[0] == '/' ||
-      (path[0] == '.' && (path[1] == '/' ||
-                          (path[1] == '.' && path[2] == '/')))) {
-    return stat64x(path, stat);
+  if (path == nullptr)
+    return -1;
+
+  char *path2 = strdup (path);
```

Comment:
Please use os::strdup and os::free. If you really intent on using the plain libc versions, use `::strdup` and `::free` to make sure - and indicate to code readers - you use the global libc variants.